core/src/main/java/com/klaytn/caver/transaction/type/EthereumAccessList.java

```diff
@@ -0,0 +1,541 @@
+        }
+
+        if(!this.getFrom().toLowerCase().equals(keyring.getAddress().toLowerCase())) {
+            throw new IllegalArgumentException("The from address of the transaction is different with the address of the keyring to use");
+        }
+
+        this.fillTransaction();
+
+        String hash = signer.apply(this);
+        SignatureData signatureData = keyring.ecsign(hash, AccountKeyRoleBased.RoleGroup.TRANSACTION.getIndex(), index);
+
+        this.appendSignatures(signatureData);
+
+        return this;
+    }
+
+    /**
+     * Getter function for to
+     *
+     * @return String
+     */
+    public String getTo() {
+        return to;
+    }
+
+    /**
+     * Setter function for to
+     *
+     * @param to The account address that will receive the transferred value.
+     */
+    public void setTo(String to) {
+        // "to" field in EthereumAccessList allows null
+        if (to == null || to.isEmpty()) {
+            to = "0x";
+        }
+
+        if (!to.equals("0x") && !Utils.isAddress(to)) {
+            throw new IllegalArgumentException("Invalid address. : " + to);
+        }
+        this.to = to;
+    }
+
+    /**
+     * Getter function for input
+     *
+     * @return String
+     */
+    public String getInput() {
+        return input;
+    }
+
+    /**
+     * Setter function for input
+     *
+     * @param input Data attached to the transaction.
+     */
+    public void setInput(String input) {
+        if (input == null) {
+            throw new IllegalArgumentException("input is missing");
+        }
+
+        if (!Utils.isHex(input)) {
+            throw new IllegalArgumentException("Invalid input : " + input);
+        }
+        this.input = Numeric.prependHexPrefix(input);
+    }
+
+
+    /**
+     * Getter function for value
+     *
+     * @return String
+     */
+    public String getValue() {
+        return value;
+    }
+
+    /**
+     * Setter function for value
+     *
+     * @param value The amount of KLAY in peb to be transferred.
+     */
+    public void setValue(String value) {
+        if (value == null) {
+            throw new IllegalArgumentException("value is missing");
+        }
+
+        if (!Utils.isNumber(value)) {
+            throw new IllegalArgumentException("Invalid value : " + value);
+        }
+        this.value = value;
+    }
+
+    /**
+     * Setter function for value
+     *
+     * @param value The amount of KLAY in peb to be transferred.
+     */
+    public void setValue(BigInteger value) {
+        setValue(Numeric.toHexStringWithPrefix(value));
+    }
+
+    /**
+     * Getter function for accessList
+     *
+     * @return accessList Access list is an EIP-2930 access list.
+     */
+    public AccessList getAccessList() {
+        return accessList;
+    }
+
+    /**
+     * Setter function for accessList
+     *
+     * @param accessList Access list is an EIP-2930 access list.
+     */
+    public void setAccessList(AccessList accessList) {
```

Comment:
Can we override the `setSignatures` function to validate `v` value?

Reply:
I think signature related logic should be handled in SignatureData class. 🤔

Reply:
Um? thats depends on transaction type, so i think that logic should be existed in Transaction.

Reply:
Please take a look [this commit](https://github.com/klaytn/caver-java/pull/291/commits/97d7670d8d77ee71d07b43ba5966d6d3b74b0094#diff-754a45f06d48ebbc479923292ba6e5fc22c9b3f5f6a9087d457c3fafdb62ce95)
and [test code](https://github.com/klaytn/caver-java/pull/291/commits/97d7670d8d77ee71d07b43ba5966d6d3b74b0094#diff-754a45f06d48ebbc479923292ba6e5fc22c9b3f5f6a9087d457c3fafdb62ce95R1106-R1152).

Reply:
For now, i will approve with those changes, but i wanna modify signature setter logic in the future!
Thank you for changing.